__tests__/run-all-tests.mjs

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env node
+
+/**
+ * Comprehensive Test Runner
+ *
+ * Runs all test suites in sequence and provides a summary report.
+ * This script helps validate the entire API after changes.
+ *
+ * Usage: npm run test:all
+ */
+
+import { exec } from 'child_process'
+import { promisify } from 'util'
+
+const execAsync = promisify(exec)
+
+// ANSI color codes for better output
+const colors = {
+	green: '\x1b[32m',
+	red: '\x1b[31m',
+	yellow: '\x1b[33m',
+	blue: '\x1b[34m',
+	cyan: '\x1b[36m',
+	magenta: '\x1b[35m',
+	reset: '\x1b[0m',
+	bold: '\x1b[1m',
+}
+
+class TestRunner {
```

Comment:
This `TestRunner` class seems very similar to this [TestSuite](https://github.com/hashicorp/web-unified-docs/pull/868/files) class. Is there anyway we can combine them?